Native/Text/src/ARText.tsx

```diff
@@ -0,0 +1,152 @@
+            const newMesh = MeshBuilder.CreateText(
+                "myText",
+                props.mxText.value ?? "",
+                font,
+                {
+                    resolution: 64,
+                    depth: 10
+                },
+                scene
+            );
+            if (newMesh) setMesh(newMesh);
+        }
+    }, [props.mxText, font]);
+
+    return (
+        <MeshComponent
+            rootMesh={mesh}
+            allMeshes={mesh ? [mesh] : undefined}
+            mxPositionType={props.mxPositionType}
+            mxPositionXStat={props.mxPositionXStat}
+            mxPositionYStat={props.mxPositionYStat}
+            mxPositionZStat={props.mxPositionZStat}
+            mxPositionYAtt={props.mxPositionYAtt}
+            mxPositionXAtt={props.mxPositionXAtt}
+            mxPositionZAtt={props.mxPositionZAtt}
+            mxPositionXExpr={props.mxPositionXExpr}
+            mxPositionYExpr={props.mxPositionYExpr}
+            mxPositionZExpr={props.mxPositionZExpr}
+            mxRotationType={props.mxRotationType}
+            mxRotationXStat={props.mxRotationXStat}
+            mxRotationYStat={props.mxRotationYStat}
+            mxRotationZStat={props.mxRotationZStat}
+            mxRotationXAtt={props.mxRotationXAtt}
+            mxRotationYAtt={props.mxRotationYAtt}
+            mxRotationZAtt={props.mxRotationZAtt}
+            mxRotationXExpr={props.mxRotationXExpr}
+            mxRotationYExpr={props.mxRotationYExpr}
+            mxRotationZExpr={props.mxRotationZExpr}
+            mxScaleType={props.mxScaleType}
+            mxScaleXStat={props.mxScaleXStat}
+            mxScaleYStat={props.mxScaleYStat}
+            mxScaleZStat={props.mxScaleZStat}
+            mxScaleXAtt={props.mxScaleXAtt}
+            mxScaleYAtt={props.mxScaleYAtt}
+            mxScaleZAtt={props.mxScaleZAtt}
+            mxScaleXExpr={props.mxScaleXExpr}
+            mxScaleYExpr={props.mxScaleYExpr}
+            mxScaleZExpr={props.mxScaleZExpr}
+            OnSceneLoaded={handleSceneLoaded}
+            mxMaterialColor={props.mxMaterialColor.value ?? "#0CABF9"}
+            mxMaterialOption={props.mxMaterialOption}
+            texture={texture}
+            mxMetalness={props.mxMetalness}
+            mxOpacity={props.mxOpacity}
+            mxRoughness={props.mxRoughness}
+            mxLightingType={props.mxLightingType}
+            mxUseDraggingInteraction={props.mxUseDraggingInteraction}
+            mxUsePinchInteraction={props.mxUsePinchInteraction}
+            mxDraggingEnabled={props.mxDraggingEnabled.value ?? false}
+            mxDragType={props.mxDragType}
+            mxOnDrag={props.mxOnDrag}
+            mxPinchEnabled={props.mxPinchEnabled.value ?? false}
+            mxPinchToScaleEnabled={false}
+            mxOnPinchActionValue={props.mxOnPinchActionValue}
+            mxOnHoverEnter={props.mxOnHoverEnter}
+            mxOnHoverExit={props.mxOnHoverExit}
+            mxOnClick={props.mxOnClick}
```

Comment:
@HedwigAR is just a NIT, but why you never use prop destructuring?
`{ ...props }`? I see here _a lot_ of repetition.

Reply:
Preference, I've done that but then I just get a massive list at the top with all these props. I could do it here again though. I go back and forth on it.